Source/Core/DolphinQt/Debugger/GekkoSyntaxHighlight.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include <QSyntaxHighlighter>
+
+#include "Common/Assembler/AssemblerShared.h"
+
+enum class HighlightFormat
+{
+  Directive,
+  Mnemonic,
+  Symbol,
+  Immediate,
+  GPR,
+  FPR,
+  SPR,
+  CRField,
+  CRFlag,
+  Str,
+  HaLa,
+  Paren,
+  Default,
+  Comment,
+  Error,
+};
+
+struct BlockInfo : public QTextBlockUserData
+{
+  std::vector<std::tuple<int, int, HighlightFormat>> block_format;
+  std::vector<std::pair<int, int>> parens;
+  std::optional<Common::GekkoAssembler::AssemblerError> error;
+  bool error_at_eol = false;
+
+  virtual ~BlockInfo() {}
+};
+
+class GekkoSyntaxHighlight : public QSyntaxHighlighter
+{
+  Q_OBJECT;
+
+public:
+  GekkoSyntaxHighlight(QTextDocument* document) : QSyntaxHighlighter(document), m_mode(0) {}
```

Comment:
m_mode can be initialized directly. m_cursor_loc should also have a default initial value.